src/nlsolve/nlsolve.jl

```diff
@@ -18,6 +18,10 @@ function nlsolve!(nlsolver::AbstractNLSolver, integrator, cache=nothing, repeat_
     else
       γW = nlsolver.γ * integrator.dt / nlsolver.α
     end
+
+    # This is for numerical differentiation cache correctness
+    # Requires Newton methods are FSAL
+    nlsolver.cache.du1 .= integrator.fsalfirst
```

Comment:
Right, all the methods are programmatically FSAL for dense outputs, we should use it well. BTW, can we simply set `integrator.fsalfirst === nlsolver.cache.du1`? Will that lead to incorrectness?

Reply:
`du1` can get overwritten/changed, for example in the central differencing method, so it's not always correct.